niap-cc/Permissions/Tester/app/src/main/java/com/android/certifications/niap/permissions/InternalPermissionTester.java

```diff
@@ -135,8 +154,165 @@ public InternalPermissionTester(TestConfiguration configuration, Activity activi
+                                    }
+                                };
+                        CompanionDeviceManager companionDeviceManager = mActivity.getSystemService(
+                                CompanionDeviceManager.class);
+
+                        companionDeviceManager.associate(request, callback, null);
+                    }
+
+                }));
+
+        mPermissionTasks.put(permission.ACCESS_AMBIENT_CONTEXT_EVENT,
+                new PermissionTest(false, Build.VERSION_CODES.TIRAMISU, () -> {
+
+                    @SuppressLint("WrongConstant") Object ambientContextManager
+                            = mContext.getSystemService("ambient_context");
+                            //Context.AMBIENT_CONTEXT_SERVICE);
+
+                    int[] eventsArray = new int[] {-1};//AmbientContextEvent.EVENT_COUGH
+                    Set<Integer> eventTypes = Arrays.stream(eventsArray).boxed().collect(
+                            Collectors.toSet());
+
+                    invokeReflectionCall(ambientContextManager.getClass(),
+                            "queryAmbientContextServiceStatus",
+                            ambientContextManager, new Class[]{Set.class, Executor.class,
+                                    Consumer.class}, eventTypes, new Executor() {
+                                @Override
+                                public void execute(Runnable runnable) {
+
+                                }
+                            }, null);
+
                 }));
+
+        // # Skip READ_ASSISTANT_APP_SEARCH_DATA
+        //Reason : Need specific role to test this permission.
+
+        // # Skip READ_HOME_APP_SEARCH_DATA
+        //Reason : Need specific role to test this permission.
+
+        mPermissionTasks.put(permission.MANAGE_SAFETY_CENTER,
+                new PermissionTest(false, Build.VERSION_CODES.TIRAMISU,() -> {
+                    mTransacts.invokeTransact(Transacts.SAFETY_CENTER_MANAGER_SERVICE,
+                            Transacts.SAFETY_CENTER_MANAGER_MANAGER_DESCRIPTOR,
+                            Transacts.getSafetyCenterConfig);
+                }));
+
+        //The permission is moved from signature permission (since android s)
+        mPermissionTasks.put(SignaturePermissions.permission.MANAGE_SENSOR_PRIVACY,
+                new PermissionTest(false, Build.VERSION_CODES.S,() -> {
+                    // ISensorPrivacyManager#setSensorPrivacy
+                    mTransacts.invokeTransact(Transacts.SENSOR_PRIVACY_SERVICE,
+                            Transacts.SENSOR_PRIVACY_DESCRIPTOR,
+                            Transacts.setSensorPrivacy, false);
+                }));
+
+        //The permission is moved from signature permission (since android t)
+        mPermissionTasks.put(SignaturePermissions.permission.TOGGLE_AUTOMOTIVE_PROJECTION,
+                new PermissionTest(false, Build.VERSION_CODES.TIRAMISU,
+                        Build.VERSION_CODES.S_V2, () -> {
```

Comment:
Won't specifying a maxSdkVersion of S_V2 prevent this test from running since it was moved from a signature permission since Android T? If the device is Android T or later, won't `BasePermissionTester.runPermissionTest` see that the device version is later than the max version of S_V2 and skip the test?

Reply:
checked

Reply:
Update looks good to me; thanks!